io.c

```diff
@@ -5030,12 +5029,6 @@ fptr_finalize_flush(rb_io_t *fptr, int noraise, int keepgvl,
     fptr->stdio_file = 0;
     fptr->mode &= ~(FMODE_READABLE|FMODE_WRITABLE);
 
-    // Ensure waiting_fd users do not hit EBADF.
-    if (busy) {
-        // Wait for them to exit before we call close().
-        do rb_thread_schedule(); while (!list_empty(busy));
```

Comment:
Is this no longer needed?
Do callers all make sure the waiting list for that IO is empty before calling `fptr_finalize_flush`?
And also that no other thread can start a blocking operation between being notified and reaching here?
`fd = -1` is only set just above but so after notification, hence it seems possible for other threads to restart their blocking operation, unless we have some flag to prevent that?

Reply:
I'm probably going to introduce a flag because the fiber scheduler can defer `io_close` operation using `io_uring` for better performance, but we can't set `fd = -1` to mean "we are closing" because then `fd` has no meaning after that and can't be used by `io_close` hook.